### PR TITLE
Adds balloon messages when holding firelocks open

### PR DESCRIPTION
**Pull request by norill**
## About The Pull Request
Adds balloon messages when a player is starts or ends holding a firelock open. Changes the examine text of a closed firelock to explain that right-click opens it permanently and left-click opens it temporarily 

## Why It's Good For The Game
Holding firelocks open behavior introduced in #57943 is very confusing. It just looks like an annoying bug, until you examine the firelock to learn that it's actually a feature. This change makes it less confusing by telling the player wtf is happening.

## Changelog
:cl:
qol: Made the action of holding firelocks open by left clicking with a crowbar less confusing by adding balloon messages describing what is happening. Changed the firelock examine text to explain that right-click opens it permanently and left-click opens it temporarily
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
Can you make this use [balloon alerts](https://github.com/tgstation/tgstation/pull/59130)? I'd like to put less and less information in the chat, especially stuff like this where people might not even see it in a chaotic moment with fire locks.

There doesn't exist yet a proc to make it like `visible_message` where everyone around you can see it, but that should be easily doable. If not, I can PR it to your branch.

**Comment by vincentiusvin** [comment]
I did not realize how messy the code was. Good job on the cleanup!

**Comment by norill** [comment]
> 
> 
> Can you make this use [balloon alerts](https://github.com/tgstation/tgstation/pull/59130)? I'd like to put less and less information in the chat, especially stuff like this where people might not even see it in a chaotic moment with fire locks.
> 
> There doesn't exist yet a proc to make it like `visible_message` where everyone around you can see it, but that should be easily doable. If not, I can PR it to your branch.

sure, why not. you could make a proc that would show a balloon to everyone, that would be really useful in many situations. consider using the same method signature as `visible_message` for consistency, that would also make it possible to use that balloon proc as a drop-in replacement for `visible_message` where it makes sense. 

one remark on the balloons themselves tho - i think their speed should vary with text length. if used with longer texts they could disappear faster than the user is able to read them, which is kinda frustrating. especially if the text was kinda important or the action isn't easily repeatable.

**Comment by Mothblocks** [comment]
Yeah I think that's fair enough, I was planning on ironing out whatever came up in more usages, but out of scope for this PR.

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -36,8 +36,8 @@
 		. += "<span class='notice'>It is open, but could be <b>pried</b> closed.</span>"
 	else if(!welded)
 		. += "<span class='notice'>It is closed, but could be <b>pried</b> open.</span>"
-		. += "<span class='notice'>Hold the door open by prying it with <i>left-click</i> and standing next to it.</span>"
-		. += "<span class='notice'>Prying by <i>right-clicking</i> the door will simply open it.</span>"
+		. += "<span class='notice'>Hold the firelock temporarily open by prying it with <i>left-click</i> and standing next to it.</span>"
+		. += "<span class='notice'>Prying by <i>right-clicking</i> the firelock will open it permanently.</span>"
```
Comment (LemonInTheDark) [inline code/game/machinery/doors/firedoor.dm:40]: This is a bit of a mis-characterization, right click doesn't perma open it, it just leaves it at the mercy of the area based firelock timer

**Comment by Mothblocks** [comment]
I'm going to make a balloon alert to viewers proc, I'll protect this from going stale in the meantime.

File: code/game/machinery/doors/firedoor.dm (hunk 1 above)
Comment (norill) [reply code/game/machinery/doors/firedoor.dm:40]: i dunno if there is a better way to word it. here "permanently" mean "not temporarily". in the grand scheme of things nothing is permanent 🤔

**Comment by Mothblocks** [comment]
That PR got merged.

**Comment by Mothblocks** [comment]
Can you update your title and changelog to match?

**Comment by norill** [comment]
sure

**Pull request merged**